client_inventory.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+
+	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
+)
+
+type InventoryReporter interface {
```

Comment:
In fact this is already messy with the client interfaces. We have a `StatusReporter` in `client_status.go`, `LogUploader` in `client_log.go`, `AuthRequester` in `client_auth.go` and so on. In fact all are doing the same thing: they are formating request are sending it to the server. 
The idea of interfaces should be to have one (maybe `Requester`) and use it interchangeably in the client code. Something to think of, but this will be rather bigger refactoring.